### PR TITLE
Use netcdftime

### PR DESCRIPTION
**Pull request by ocefpaf**
This PR uses the new [`netcdftime` package](https://github.com/Unidata/netcdftime) instead of `netCDF4-python`.

I can rebase this later but I left the individual commits to make it easier to review.

There are 6 tests failing but I believe they are unrelated to this PR b/c they are failing on master too. Note that on master py34 was passing, but I guess that is due to an old `netCDF4-python` version.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@ language: python
 
 python:
   - 2.7
-  - 3.4
```
Comment (ocefpaf) [inline .travis.yml]: conda-forge deprecated py34

File: .travis.yml [hunk 2]
```diff
@@ -19,19 +19,13 @@ install:
...
   # Create the basic testing environment
   # ------------------------------------
-  - conda config --set always_yes yes --set changeps1 no
-  - conda config --set show_channel_urls True
+  - conda config --set always_yes yes --set changeps1 no --set show_channel_urls True
   - conda update --quiet conda
+  - conda config --add channels conda-forge
   - ENV_NAME='test-environment'
-  - conda create --quiet -n $ENV_NAME python=$TRAVIS_PYTHON_VERSION
+  - conda create --name $ENV_NAME python=$TRAVIS_PYTHON_VERSION udunits2 --file requirements.txt --file requirements-dev.txt
```
Comment (ocefpaf) [inline .travis.yml:25]: Building the env in one go saves us some trouble. Sometimes `conda` does not respect channel preference when adding packages to an env.

File: setup.py [hunk 3]
```diff
@@ -2,21 +2,9 @@
 import sys
 
 from setuptools import setup
-from setuptools.command.test import test as TestCommand
 import versioneer
 
 
-class PyTest(TestCommand):
```
Comment (ocefpaf) [inline setup.py]: I could not get this to work, kept getting "tests not found." I switched to a direct `pytest` call with the `pytest-cov` plugin instead, hope that is OK.

**Comment by pelson** [comment]
Looks like some genuine test failures here. Happy in general with the changes. Thanks @ocefpaf!

**Comment by ocefpaf** [comment]
> Looks like some genuine test failures here.

I'll take a closer look at those soon.

> Happy in general with the changes.

Maybe we should wait for a stable release of `cftime` and/or the next release of `netCDF4`, which will be using that. The package is not stable yet and they just renamed it.

**Comment by ocefpaf** [comment]
@pelson the failures seems to be due to a change in how the comparison works. It is refusing compare dates with different calendars:

```
TypeError: cannot compare cftime._cftime.Datetime360Day(1970, 1, 1, 0, 0, 20, 0, -1, 1) and cftime._cftime.datetime(1970, 1, 1, 0, 0, 20, 0, -1, 1) (different calendars)
```

That makes sense to me but I'm not sure how to proceed to make the tests valid again. Maybe we can compare the timetuple like

```python
cftime.datetime(1970, 1, 1, 0, 0, 20).timetuple() == _num2date_to_nearest_second(20., useconds).timetuple()
True
```

Is that OK?

**Comment by pelson** [comment]
> Maybe we should wait for a stable release of cftime and/or the next release of netCDF4, which will be using that. The package is not stable yet and they just renamed it.

That works for me.

> the failures seems to be due to a change in how the comparison works. It is refusing compare dates with different calendars:

In which case, cf_units is testing questionable behaviour IMO. I'm fully supportive of removing any such behaviour, unless there was a genuine mistake in writing the test (I didn't look yet).

File: .travis.yml [hunk 4]
```diff
@@ -18,18 +18,13 @@ install:
 
   # Create the basic testing environment
   # ------------------------------------
-  - conda config --set always_yes yes --set changeps1 no
-  - conda config --set show_channel_urls True
+  - conda config --set always_yes yes --set changeps1 no --set show_channel_urls True
   - conda update --quiet conda
+  - conda config --add channels conda-forge
```
Comment (ocefpaf) [inline .travis.yml:23]: conda-forge's `udunits2` is newer, hope that this change is OK @bjlittle

**Comment by bjlittle** [comment]
@ocefpaf Could you rebase, now that #91 has been merged, thanks!

**Comment by ocefpaf** [comment]
> @ocefpaf Could you rebase, now that #91 has been merged, thanks!

Already did :smile:

**Comment by bjlittle** [comment]
@ocefpaf The `coveralls` coverage has tanked... perhaps due to the introduction of `cftime`?

Could you update the `.coveragerc` file to also `omit` the  `.eggs` directory i.e.

```shell
[run]
branch = True
omit =
    setup.py
    versioneer.py
    cf_units/_version.py
    cf_units/etc/*
    cf_units/tests/*
    .eggs
```
That should do the trick...

**Comment by bjlittle** [comment]
@ocefpaf Let's do that rebase tango again... :wink:

**Comment by ocefpaf** [comment]
> @ocefpaf Let's do that rebase tango again... :wink:

Done!

**Comment by bjlittle** [comment]
@ocefpaf Awesome, thanks :+1:

**Pull request merged**